lib/model.dart

```diff
@@ -1,29 +1,30 @@
 import 'package:objectbox/objectbox.dart';
 import 'package:hive/hive.dart';
+import 'package:isar/isar.dart' as isar;
 
 part 'model.g.dart';
 
+@isar.Collection()
 @Entity()
 @HiveType(typeId: 1)
 class TestEntity {
+  @isar.Id()
   @Id()
   @HiveField(0)
-  int id;
+  late int id;
 
   @HiveField(1)
-  String tString;
+  late String tString;
 
   @Property(type: PropertyType.int)
   @HiveField(2)
-  int tInt; // 32-bit
+  late int tInt; // 32-bit
 
   @HiveField(3)
-  int tLong; // 64-bit
+  late int tLong; // 64-bit
 
   @HiveField(4)
-  double tDouble;
-
-  TestEntity(this.id, this.tString, this.tInt, this.tLong, this.tDouble);
```

Comment:
maybe leave as is and add another public constructor for Isar (only needed until they support `final` fields, which I'm sure they would - that would basically mean it has to recognize constructors).

This should work:
```dart
  // Isar v0.4: Class needs to have a public zero-arg constructor.
  TestEntity.forIsar()
      : id = 0,
        tString = '',
        tInt = 0,
        tLong = 0,
        tDouble = 0;
```

Reply:
Yes `final` field support is incoming I believe.
The proposed solution for the constructor _almost_ works. The generator does not pick up the zero arg constructor and instead uses the Class name to generate the Adapter. This is a bug of course.

FYI to run this PR, after the build_runner step is done, manually edit the `isar.g.dart` to use the `forIsar` constructor in the `deserialize` method.

I will wait for a while until these issues are resolved.